zhaquirks/tuya/ts0601_switch.py

```diff
@@ -748,3 +748,293 @@ class TuyaSextupleSwitchTO_GP(TuyaSwitch):
             },
         }
     }
+
+
+class TuyaSwitchX8_GP(TuyaSwitch):
+    """Tuya x8 channels switch with GreenPowerProxy cluster device."""
+
+    signature = {
+        MODELS_INFO: [
+            ("_TZE200_wktrysab", "TS0601"),
+        ],
+        ENDPOINTS: {
+            # <SimpleDescriptor endpoint=1 profile=260 device_type=81
+            # input_clusters=[0x0000,0x0004,0x0005,0xEF00]
+            # output_clusters=[0x000A,0x0019]>
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.SMART_PLUG,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    Groups.cluster_id,
+                    Scenes.cluster_id,
+                    TuyaOnOffManufCluster.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [Time.cluster_id, Ota.cluster_id],
+            },
+            # <SimpleDescriptor endpoint=242 profile=41440 device_type=97
+            # input_clusters=[]
+            # output_clusters=[33]
+            242: {
+                PROFILE_ID: zgp.PROFILE_ID,
+                DEVICE_TYPE: zgp.DeviceType.PROXY_BASIC,
+                INPUT_CLUSTERS: [],
+                OUTPUT_CLUSTERS: [GreenPowerProxy.cluster_id],
+            },
+        },
+    }
+
+    replacement = {
+        ENDPOINTS: {
+            1: {
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_LIGHT,
```

Comment:
I guess it's currently being done for all TS0601 switches, but any other reasons why we're replacing the `SMART_PLUG` type with `ON_OFF_LIGHT` (everywhere)?

Reply:
The short answer is because I do a lot of copy&paste.

The long answer would be that most (all?) of the TS0601 devices are defined as `SMART_PLUG` (sensors, TRV, valves...). So redefine the `DEVICE_TYPE` is normal for the TS0601. Probably when firsts switches where done there was no easy way to do the change in HA or we were not aware, and to define switches as lights is a common mistake (to define for what the device is used and not what it really is). 

I think I've seen some recent conversation about it and I totally agree with redefining the device from HA and not using a quirk just to modify it.

In this specific case (TS0601 devices) I do not think that the DEVICE_TYPE can be maintained in all cases (all kind of devices).

Therefore, the question to be resolved would be how do we want to manage the TS0601 switches?
Since Tuya's design seems wrong, IMHO the switches quirks should change the DEVICE_TYPE, what I'm not clear about is which one to choose. The current `ON_OFF_LIGHT` doesn't seem correct to me. Depending on the device I would opt for `ON_OFF_SWITCH` or `LEVEL_CONTROL_SWITCH` for the dimmable type.

Reply:
> Since Tuya's design seems wrong, IMHO the switches quirks should change the DEVICE_TYPE, what I'm not clear about is which one to choose. The current ON_OFF_LIGHT doesn't seem correct to me. Depending on the device I would opt for ON_OFF_SWITCH or LEVEL_CONTROL_SWITCH for the dimmable type.

Yeah. We likely still want the "dimmable light" device type for all "dimmers", as HA doesn't support dimming switches.
So the question for the device type only occurs for on/off switches/relays.

At the moment, most(?) other Tuya quirks for switches with relays seem to use `ON_OFF_LIGHT` which isn't ideal like you mentioned. (But we can't change existing quirks without breaking existing setups.)

So for this quirk, we could either go with `ON_OFF_LIGHT` (which aligns with most other Tuya switch quirks), but doesn't seem entirely correct, or use `ON_OFF_SWITCH` like you mentioned (which would create a more correct switch entity in HA) (but then not really align with the other Tuya quirks).

So, what do you like better? 😄 
I think I'd prefer doing it "right from here" and use `ON_OFF_SWITCH` in this case.

Reply:
Yeah, I also prefer use the `ON_OFF_SWITCH`.
Tested in https://github.com/zigpy/zha-device-handlers/issues/2492#issuecomment-1745851187